core/lib/zksync_core/src/metadata_calculator/updater.rs

```diff
@@ -146,40 +143,42 @@ impl TreeUpdater {
-                        .blocks_dal()
-                        .get_events_queue(header.number)
-                        .await
-                        .unwrap()
-                        .unwrap();
-                    let events_queue_commitment = events_queue_commitment_fixed(&events_queue);
-                    events_queue_commitment_latency.observe();
-
-                    let bootloader_commitment_latency =
-                        METRICS.start_stage(TreeUpdateStage::BootloaderCommitment);
-                    let initial_bootloader_contents = storage
-                        .blocks_dal()
-                        .get_initial_bootloader_heap(header.number)
-                        .await
-                        .unwrap()
-                        .unwrap();
-                    let full_bootloader_memory =
-                        expand_bootloader_contents(&initial_bootloader_contents);
-                    let bootloader_initial_content_commitment =
-                        initial_heap_content_commitment_fixed(&full_bootloader_memory);
-                    bootloader_commitment_latency.observe();
-
-                    (
-                        Some(H256(events_queue_commitment)),
-                        Some(H256(bootloader_initial_content_commitment)),
-                    )
-                } else {
-                    (None, None)
-                };
+            let (events_queue_commitment, bootloader_initial_content_commitment) = if self.mode
+                == MerkleTreeMode::Full
+            {
+                let events_queue_commitment_latency =
+                    METRICS.start_stage(TreeUpdateStage::EventsCommitment);
+                let events_queue = storage
+                    .blocks_dal()
+                    .get_events_queue(header.number)
+                    .await
+                    .unwrap()
+                    .unwrap();
+                let events_queue_commitment =
+                    events_queue_commitment(&events_queue, header.protocol_version.unwrap());
+                events_queue_commitment_latency.observe();
+
+                let bootloader_commitment_latency =
+                    METRICS.start_stage(TreeUpdateStage::BootloaderCommitment);
+                let initial_bootloader_contents = storage
+                    .blocks_dal()
+                    .get_initial_bootloader_heap(header.number)
+                    .await
+                    .unwrap()
+                    .unwrap();
+                let bootloader_initial_content_commitment = bootloader_initial_content_commitment(
+                    &initial_bootloader_contents,
+                    header.protocol_version.unwrap(),
+                );
+                bootloader_commitment_latency.observe();
+
+                (
+                    events_queue_commitment,
+                    bootloader_initial_content_commitment,
+                )
+            } else {
+                (None, None)
+            };
```

Comment:
does it make sense to move the whole logic to the commitment_utils?

Reply:
hmm, I don't think it's a good idea for `commitment_utils` to depend on DAL. I'd rather extract this part to a separate function

Reply:
sorry, forget about it..